system/src/system_task.cpp

```diff
@@ -467,6 +467,26 @@ void manage_ble_prov_mode() {
 #endif
 }
 
+extern "C" void system_isr_task_queue_free_memory(uint8_t *ptrToFree) {
```

Comment:
This is going to be unused on all the other platforms except for rtl872x-based which is going to cause the compiler to complain that this is unused. Perhaps move into rtl872x-specific source files under HAL? Or perhaps let's do a proper thing and add this function into the system_task.h header, implement it here, disable unused warning for it.